### PR TITLE
Clean up all compilation warnings

### PR DESCRIPTION
**Pull request by juanjoDiaz**
No functional changes.

Just clean up al compilation warning:
* Unused imports
* Clases that could be lambdas
* Deprecated methods
* etc.

### CODE REVIEW

File: awslambda/src/main/scala/akka/stream/alpakka/awslambda/scaladsl/AwsLambdaFlow.scala [hunk 1]
```diff
@@ -8,7 +8,7 @@ import akka.NotUsed
 import akka.stream.alpakka.awslambda.AwsLambdaFlowStage
 import akka.stream.scaladsl.Flow
 import com.amazonaws.services.lambda.model.{InvokeRequest, InvokeResult}
-import com.amazonaws.services.lambda.{AWSLambdaAsyncClient, AWSLambdaClient}
+import com.amazonaws.services.lambda.{AWSLambdaAsyncClient}
```
Comment (raboof) [inline awslambda/src/main/scala/akka/stream/alpakka/awslambda/scaladsl/AwsLambdaFlow.scala]: could remove the `{}`

**Comment by raboof** [comment]
Travis is reporting compilation failures though, perhaps missed something there?

**Comment by ennru** [comment]
A few types go wrong...

**Comment by juanjoDiaz** [comment]
It seems that scala 2.11 is not very happy with lambdas and functional interfaces....
I'll rollback those.

**Comment by ennru** [comment]
Thank you.

**Comment by 2m** [comment]
This is great! We could also enable critical warnings so no warnings creep up into the codebase from now on.

**Pull request merged**